firedrake/mesh.py

```diff
@@ -2921,13 +2913,13 @@ def VertexOnlyMesh(mesh, vertexcoords, missing_points_behaviour='error',
                 warn(str(error))
             else:
                 raise ValueError("missing_points_behaviour must be IGNORE, ERROR or WARN")
-
     name = name if name is not None else mesh.name + "_immersed_vom"
-
+    swarm.setName(_generate_default_mesh_topology_name(name))
+    original_swarm.setName(_generate_default_mesh_topology_name(name) + "_input_ordering")
```

Comment:
Is this now consistent with DMPlex? Do they have the same name as their mesh?

Reply:
`DMSwarm` and `DMPlex` are both `DM`, `DM` is `PetscObject`, and `PetscObject` carries `setName()` method.
Could you clarify your question?

Reply:
Upon construction of a normal mesh, you can supply a name string. Do you use this string (or a modification of it for, say, extruded meshes) to name the `DMPlex` `PetscObject` associated with that mesh?

Reply:
Yes. A `DMPlex` has the same name as the `MeshTopology`.